File: CMEEminiPrject/code/tmp.py
# load modules
import pandas as pd
import  scipy as sc
import matplotlib.pylab as pl
import seaborn as sns
from lmfit import Model

# import data
data = pd.read_csv("../data/CRat.csv")
# print("Loaded{}columns.".format(len(data.columns.values)))
# data.head()

# print(data.columns.values)
# print(data.TraitUnit.unique())
# print(data.ResDensityUnit.unique())
# print(data.ID.unique())

##### define the models:
## Holling models:
# def holling(Xr, a, h, c):
#     """
#     Holling model description
#     """
#     model = a*Xr/(1+h*a*Xr)
#     residual = model - c
#     return residual
def holling(Xr, a, h):
    """
    Holling model description
    """
    c = a*Xr/(1+h*a*Xr)
    return c

def genholling(Xr, a, h, q):
    """
    Holling generalised model
    """ 
    c = a*Xr^(q+1)/(1+h*a*Xr^(q+1))
    return c
# data[data["ID"] == 39835]

# data.ResDensity[data["ID"] == 39835]


##### model fitting:
hollmodel= Model(holling)

parameters = hollmodel.fit(data.N_TraitValue, Xr = data.ResDensity_SI_VALUE, a = 0.1,h= 1e-05)
parameters.fit_report()

for i in data.ID.unique():
    
    data[data['ID'] == i]
    y = data.N_TraitValue[data['ID'] == i]
    x = data.ResDensity_SI_VALUEdata[data['ID'] == i]
    parameters = hollmodel.fit(y, x = x, a = ,h= )


data_sub_39982 = data[data["ID"]== 39982]
data_sub_39982.head

File: CMEEminiPrject/code/tmp.R
#### load packages

require(repr)
require(minpack.lm)

#### import data
data = read.csv("../data/CRat.csv", header = TRUE)
colnames(data)
# wranged = subset(data,)

sub= subset(data, ID=="687")
#### define models
## linear models:
lms=lm(TraitValue~ResDensity_SI_VALUE, data = subs)
  for( i in unique(data[,1])){
    subs = subset(data, ID == i)
    lms=lm(TraitValue~ResDensity_SI_VALUE, data = subs)
    plms= lm(TraitValue~poly(ResDensity_SI_VALUE,2), data = subs)
    cubic= try(lm(TraitValue~poly(ResDensity_SI_VALUE,3), data = subs),silent = T)
    
    ## calculate and compare AICs
    if(class(cubic)!="try-error"){
      aics=as.data.frame(AIC(lms,plms,cubic))
    }else{
      aics=as.data.frame(AIC(lms,plms))
    }
    print(paste(as.character(i),row.names(aics[which(aics[,2]==min(aics[,2])),])))
  }
  

## polynomial models:
plms= lm(TraitValue~poly(ResDensity_SI_VALUE,2), data = subs)


#### starting value, need to improve



Hollfit<- nlsLM(N_TraitValue~HollingMod(ResDensity,a,h), data = data, start = list(a= .1, h=.01))

len= seq(min(data$ResDensity),max(data$ResDensity), len=2000)
coef(Hollfit)["a"]
coef(Hollfit)["h"]
fitting= HollingMod(len, coef(Hollfit)["a"], coef(Hollfit)["h"])
lines(len, fitting, col = 'red', lwd=2)
## Generalized Holling model:



aics
